plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidatorTest.java

```diff
@@ -70,18 +72,47 @@ public void tearDown() throws CoreException {
   @Test
   public void testValidate_withWrongVersion()
       throws IOException, CoreException, ParserConfigurationException {
-    byte[] bytes = XML.getBytes(StandardCharsets.UTF_8);
-    WebXmlValidator validator = new WebXmlValidator();
+    String xml = "<web-app xmlns='http://xmlns.jcp.org/xml/ns/javaee'"
+        + " version='3.1'></web-app>";
+    byte[] bytes = xml.getBytes(StandardCharsets.UTF_8);
     validator.validate(resource, bytes);
     IMarker[] markers = resource.findMarkers(SERVLET_MARKER, true, IResource.DEPTH_ZERO);
     assertEquals(1, markers.length);
     assertEquals(ELEMENT_MESSAGE, (String) markers[0].getAttribute(IMarker.MESSAGE));
   }
   
+  @Test
+  public void testValidate_withExistingServletClass()
```

Comment:
Can you add a test where the ServletClass is in a package and used as such in the web.xml? E.g. `<servlet-class>foo.bar.ServletClass</servlet-class>`